spec/requests/content_pages_spec.rb

```diff
@@ -44,4 +44,38 @@
       end
     end
   end
+
+  describe 'GET /modules/:training_module_id/content-pages/1-3-4' do
+    before do
+      get training_module_content_page_path(:alpha, '1-3-4')
+    end
+
+    context 'when module is not completed' do
+      it 'shows not completed' do
+        expect(response.body).to include('You have not yet completed the module.')
+      end
+
+      it 'does not have the users name' do
+        expect(response.body).not_to include('John Doe')
+      end
+    end
+
+    context 'when module is completed' do
+      include_context 'with progress'
+
+      before do
+        view_whole_module(alpha)
+      end
+
+      it 'shows completed' do
```

Comment:
@peterdavidhamilton @mo-zag @agoldstone93 @tunde-anddigital 

Is there an easy way to mark a module complete so the certificate will render correctly here?

Reply:
I am reviewing this today @ebrett but having worked with @agoldstone93 yesterday and with it fresh in my mind... "completeness" for module/sub-module/topic is determined by page view events. In addition, primarily for tracking, we have specifically named events which will only be recorded once.

Reply:
Example of this is being done in @agoldstone93's KPI PR using`view_whole_module` and an additional `complete_module` event.

Reply:
`completed?` is false because the events aren't there for every item

Reply:
The `user` for which the events are created is not the same one as you're signing in as in the spec.

Reply:
OK, sorted - I'm now using the `user` created in the 'with progress` context.  I needed to make them a registered user. Also needed to wait until the `view_whole_module` was done before visiting the certificate page, so the order of the before actions was important.  Specs are now passing